From: Tomasz Maciej Nowak <tmn505@gmail.com>
Subject: HACK: mtd: spi-nor: spansion: remove S25FL064K support

This chip uses the same JEDEC ID as Winbond W25Q64FV not having same
features implemented. Because of same id, spi-nor driver always recognizes
s25fl064k flash, even tough the Winbond might be installed. Hardcoding
compatible value in dts also doesn't help. So remove the conflicting chip,
because we need to set locking features for Winbond chip and we're not
able to verify if this feature is supported by Spansion chip.

Ref: https://lore.kernel.org/lkml/1450205301-32207-1-git-send-email-computersforpeace@gmail.com

--- a/drivers/mtd/spi-nor/spansion.c
+++ b/drivers/mtd/spi-nor/spansion.c
@@ -825,9 +825,6 @@ static const struct flash_info spansion_
 	{ "s25fl016k",  INFO(0xef4015,      0,  64 * 1024,  32)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
-	{ "s25fl064k",  INFO(0xef4017,      0,  64 * 1024, 128)
-		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
-			      SPI_NOR_QUAD_READ) },
 	{ "s25fl116k",  INFO(0x014015,      0,  64 * 1024,  32)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
